pkg/apis/pipeline/v1beta1/pipelinerun_conversion_test.go

```diff
@@ -143,6 +145,76 @@ func TestPipelineRunConversion(t *testing.T) {
 					},
 				},
 			},
+			Status: v1beta1.PipelineRunStatus{
+				Status: duckv1beta1.Status{
+					Conditions: duckv1beta1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: corev1.ConditionTrue,
+					}},
+				},
+				PipelineRunStatusFields: v1beta1.PipelineRunStatusFields{
+					StartTime:      &metav1.Time{Time: time.Now()},
+					CompletionTime: &metav1.Time{Time: time.Now().Add(1 * time.Minute)},
+					PipelineResults: []v1beta1.PipelineRunResult{{
+						Name: "pipeline-result-1",
+						Value: *v1beta1.NewObject(map[string]string{
+							"pkey1": "val1",
+							"pkey2": "rae",
+						})}, {
+						Name: "pipeline-result-2",
+						Value: *v1beta1.NewObject(map[string]string{
+							"pkey1": "val2",
+							"pkey2": "rae2",
+						}),
+					}},
+					PipelineSpec: &v1beta1.PipelineSpec{
+						Tasks: []v1beta1.PipelineTask{{
+							Name: "mytask",
+							TaskRef: &v1beta1.TaskRef{
+								Name: "mytask",
+							},
+						}},
+					},
+					SkippedTasks: []v1beta1.SkippedTask{
+						{
+							Name:   "skipped-1",
+							Reason: v1beta1.WhenExpressionsSkip,
+							WhenExpressions: []v1beta1.WhenExpression{{
+								Input:    "foo",
+								Operator: "notin",
+								Values:   []string{"foo", "bar"},
+							}},
+						}, {
+							Name:   "skipped-2",
+							Reason: v1beta1.MissingResultsSkip,
+						},
+					},
+					ChildReferences: []v1beta1.ChildStatusReference{
+						{
+							TypeMeta:         runtime.TypeMeta{Kind: "TaskRun"},
```

Comment:
Shall we have this kind here or that's what `typeMeta` should handle inline?

Reply:
We need to keep this

Reply:
Agreed with what @vdemeester mentioned, we'd need to convert all the `TypeMeta` **except for** those top-level CRDs (`Task`, `TaskRun`, `Pipeline`, `PipelineRun`, `Run`). The idea is those `TypeMeta` (outside of none-top-level CRDs) would be mainly for referring to a resource of a specific type and version (though there might be different case by case).